synergy-design-system/synergy-design-system
# docs: 📚 Add codepen link to stories

# Pull Request

## 📖 Description

This PR adds the ability to create a codepen from the currently viewed story. It uses the source obtained from the DOM to do this and formats it using prettier.

Current todos:

- [ ] Validate it runs with our storybook
- [ ] Make initially loaded scripts configurable (currently they are hard coded)
- [ ] Test with all possible stories to make sure it does not break
- [ ] Optional: Have a look if we can support the code in storybooks play functions somehow?
- [ ] Add eslint support
- [ ] Discuss publishing it to storybook addons when ready

### 🎫 Issues

Closes #376

## 👩‍💻 Reviewer Notes

The addons code is located in `src/packages/_private/storybook-addon-codepen` and already added as a addon in `src/packages/docs` for our own storybook instance.


## 📑 Test Plan



- Check out this branch
- Use `pnpm i && pnpm build` to create the addon
- Try out the plugin with a pristine storybook by launching it directly via `cd packages/src/_private/storybook-addon-codepen && pnpm start` **or**
- try it out by launching our own storybook.

## ✅ DoD
